LLM entries

gustavo-salazar opened the pull request:
This PR is a way to represent the LLM tagged entries.

Basically, I'm reusing the same styling as with LLM descriptions:
* The same type of badge is now displayed in the top of the summary of an entry tagged as LLM, including the `reviewed` or `unreviewed` part.
* I'm extending the Callout box, and changing the content depending if an LLM was used to generate the description, the entry details, or both.

To test this PR I deployed the `dev` API using the `dw-changes` branch: http://wp-np3-bc.ebi.ac.uk/interpro/api/

I also manually changed some of the panther entries in the DB to use them as test data:
* PTHR10000: Entry AND description AI generated unreviewed
* PTHR10003: Entry AI generated reviewed - Description imported from interpro but not LLM
* PTHR10005: Entry AI generated unreviewed - Description imported from interpro but not LLM
* PTHR10009: Description imported from interpro, which was in turn LLM-generated

typhainepl commented:
I feel like having AI generated at the top of the entry can be misleading. For example, when I see such information, my interpretation is that the model has been generated using AI, which is not the case here.
I remember we had discussed adding a label for all the fields generated by AI (i.e. entry name, short name and description) instead.

PTHR10003: Entry AI generated reviewed - Description imported from interpro but not LLM
PTHR10005: Entry AI generated unreviewed - Description imported from interpro but not LLM
=> when will we encounter such cases? Is it when the name of the signature has been generated by AI, but not the description?

gustavo-salazar commented:
> I feel like having AI generated at the top of the entry can be misleading. For example, when I see such information, my interpretation is that the model has been generated using AI, which is not the case here. I remember we had discussed adding a label for all the fields generated by AI (i.e. entry name, short name and description) instead.

Correct we discussed such an option, but we didn't agree on it, as it might create too many badges. So the version here is the simplest implementation for it, but of course, it is open to discussion. I could implement an alternative version that has a label on each field, so we can compare.


> PTHR10003: Entry AI generated reviewed - Description imported from interpro but not LLM PTHR10005: Entry AI generated unreviewed - Description imported from interpro but not LLM => when will we encounter such cases? Is it when the name of the signature has been generated by AI, but not the description?

Yes, that sort of cases, I don't really know if they would actually appear in the database, but given the returned flags, it is a possibility:  for example on `PTHR10003` the flag that an entry is LLM is `true`, but that entry doesn't have a description, so it takes it from the InterPro entry that has been integrated to. And that entry can itself have a LLM generated description, which is not the case for `PTHR10003`, but it is for `PTHR10005`

typhainepl commented:
> Correct we discussed such an option, but we didn't agree on it, as it might create too many badges. So the version here is the simplest implementation for it, but of course, it is open to discussion. I could implement an alternative version that has a label on each field, so we can compare.

I agree that having too many badges isn't ideal. Since we already have a label for the descriptions, we could change the label at the top to `AI-generated name and short name` or something similar. What do you think?

>Yes, that sort of cases, I don't really know if they would actually appear in the database, but given the returned flags, it is a possibility: for example on PTHR10003 the flag that an entry is LLM is true, but that entry doesn't have a description, so it takes it from the InterPro entry that has been integrated to. And that entry can itself have a LLM generated description, which is not the case for PTHR10003, but it is for PTHR10005

I don't think we'll encounter such cases, but using the label suggested above will help clarify.

matthiasblum reviewed (commented):
Looks good.

Regarding informing users about AI-generated info, we could simplify by showing a generic warning message under the entry's name, such as:

![image](https://github.com/ProteinsWebTeam/interpro7-client/assets/2097501/5cdc0c6d-8164-49c7-abe9-1dded7d76344)

> This entry contains information, such as name, short name, or description, that has been generated using an AI language model. Please exercise discretion when interpreting the information provided.

This message clearly informs that some of the _annotations_ have been generated using AI (and not the model itself).

If at least one of the AI-generated information is reviewed (name, short name, at least one block of the description), the message would mention that the information has not been reviewed:

> This entry includes information, such as name, short name, or descriptions, that has been generated using an AI language model and has not undergone review by expert curators. Please exercise discretion when interpreting the information provided.

I don't think that users will want to know what _exactly_ has been generated using AI (name and short name? which part of the description?), but if some want to, the information will be available through the REST API.

---

Also, the "Provide feedback" and "Add your annotation" are somehow redundant when looking at InterPro entries. We don't need to do the change as part of this PR, but eventually we could remove the link to the Google Form (and maybe change the text of "Add your annotation" to "Provide feedback").

---

> PTHR10003: Entry AI generated reviewed - Description imported from interpro but not LLM

Won't happen. We'll let curators flag an InterPro entry with AI-generated name and short name as reviewed, but not member database signatures (technically possible but complicated).

gustavo-salazar commented:
OK will implement a version with the suggestions. And will share here again for discussion.

---

@matthiasblum Some questions/comments from your feedback:

> I don't think that users will want to know what _exactly_ has been generated using AI (name and short name? which part of the description?), but if some want to, the information will be available through the REST API.

Currently, the API would identify if descriptions, or "entries" are LLM, not really identifying if it was the name, short_name, etc. Are you thinking of adding more details to the LLM flag?

> Also, the "Provide feedback" and "Add your annotation" are somehow redundant when looking at InterPro entries. We don't need to do the change as part of this PR, but eventually we could remove the link to the Google Form (and maybe change the text of "Add your annotation" to "Provide feedback").

I'll create a GH issue for this.

> > PTHR10003: Entry AI generated reviewed - Description imported from interpro but not LLM
>
> Won't happen. We'll let curators flag an InterPro entry with AI-generated name and short name as reviewed, but not member database signatures (technically possible but complicated).

I wasn't talking about AI generated signatures. My point is that when a member DB entry doesn't have a description, we import it from its interpro. (e.g. [PTHR10003](https://www.ebi.ac.uk/interpro/entry/panther/PTHR10003/)), but that description itself could have been LLM generated itself. Which I guess is useful to inform the user about it.

matthiasblum commented:
> > I don't think that users will want to know what _exactly_ has been generated using AI (name and short name? which part of the description?), but if some want to, the information will be available through the REST API.
> 
> Currently, the API would identify if descriptions, or "entries" are LLM, not really identifying if it was the name, short_name, etc. Are you thinking of adding more details to the LLM flag?

No, no. What I meant is that the API now has `is_llm` at the entry level, and `llm` at the description level. If `is_llm` is `true`, the name _and_ short name have been generated using AI. If `llm` is `true`, the description has been generated using AI. And if there is a mix of AI-generated and human-curated descriptions, it is possible to know the source/origin of each one (since there is one `llm` field per description). So if users want to know which annotation in the entry has been generated using AI, they can get this information from the API.

>> Won't happen. We'll let curators flag an InterPro entry with AI-generated name and short name as reviewed, but not member database signatures (technically possible but complicated).
>
> I wasn't talking about AI generated signatures. My point is that when a member DB entry doesn't have a description, we import it from its interpro. (e.g. [PTHR10003](https://www.ebi.ac.uk/interpro/entry/panther/PTHR10003/)), but that description itself could have been LLM generated itself. Which I guess is useful to inform the user about it.

At the moment, we use AI to generate annotations of unintegrated signatures, then we plan to auto-create InterPro entries by integrating these signatures and use the AI-generated annotations of the signature for the entry. So if an entry has a AI-generated description, it's because it has been imported from the signature. But your point stands as curators could integrate a signature (that doesn't have a description) in an entry which was originally created from another signature (that has an AI-generated description). :exploding_head:

gustavo-salazar commented:
Ok, I just did a minor version change where I removed the Badge for AI-generated Entries, and simplified the text in the callout. But given that we have the info on which field was LLM generated I added it as a Tooltip, that way the info is still there but doesn't take much space, and the message stays simple.

typhainepl reviewed (approved):
The new version looks better :)

matthiasblum commented on src/components/Entry/BadgeAI/style.css:
```diff
@@ -0,0 +1,21 @@
+.vf-badge--tertiary {
+  background: #e8e8e8;
```
```suggestion
  background: rgba(255, 220, 220, 0.5);
```

matthiasblum commented on src/components/Entry/BadgeAI/style.css:
```diff
@@ -0,0 +1,21 @@
+  background: #e8e8e8;
+  border-color: transparent;
+  border-radius: 0.3em;
+  color: rgba(0, 0, 0, 0.6);
```
```suggestion
  color: rgba(159, 58, 56, 1);
```

matthiasblum commented on src/components/Entry/BadgeAI/style.css:
```diff
@@ -0,0 +1,21 @@
+}
+.vf-badge--tertiary:hover,
+.vf-badge--tertiary:visited {
+  color: rgba(0, 0, 0, 0.6);
```
```suggestion
  color: rgba(159, 58, 56, 1);
```
